### ISSUE TITLE
MacVtap L2 Network connectivity (LLDP) only working while running tcpdump

### ISSUE THREAD

**Issue by fizzers123** [kubevirt/kubevirt#9464]
Hello KubeVirt Community 

We are trying to establish L2 connectivity between KubeVirt VMs. [MacVtap](https://kubevirt.io/user-guide/virtual_machines/interfaces_and_networks/#start-a-vm-with-macvtap-interfaces) seems like a promising option for this, as it eliminates the bridge in the virt-launcher. However, LLDP still doesn't want to work. 

The LLDP packages arrive at the physical interface but are not passed to the VM. To enable this, I think I would need to set `trustGuestRxFilters="yes"` ([discussion](https://superuser.com/questions/944678/how-to-configure-macvtap-to-let-it-pass-multicast-packet-correctly)). But I am not sure how I can enable this? 

I found it referenced here: https://github.com/kubevirt/kubevirt/blob/9d100a211ca3bfe0925137b002d8b6b4590da922/pkg/virt-launcher/virtwrap/api/schema.go#L720

<details>
  <summary>Enable Feature Gate</summary>

```yaml
---
apiVersion: kubevirt.io/v1
kind: KubeVirt
metadata:
  name: kubevirt
  namespace: kubevirt
spec:
  configuration:
    developerConfiguration: 
      featureGates:
        - Macvtap
     
```

</details>

<details>
  <summary>Install macvtap-cni</summary>

`kubectl apply -f https://raw.githubusercontent.com/kubevirt/macvtap-cni/main/manifests/macvtap.yaml`

```yaml
---
kind: ConfigMap
apiVersion: v1
metadata:
  name: macvtap-deviceplugin-config
data:
  DP_MACVTAP_CONF: '[]'
  
```

</details>

<details>
  <summary>Deploy Test VMs</summary>

```yaml
---
kind: NetworkAttachmentDefinition
apiVersion: k8s.cni.cncf.io/v1
metadata:
  name: net1
  annotations:
    k8s.v1.cni.cncf.io/resourceName: macvtap.network.kubevirt.io/ens18
spec:
  config: '{
      "cniVersion": "0.3.1",
      "name": "net1",
      "type": "macvtap",
      "mtu": 1500
    }'
---
apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachine
metadata:
  name: vm-ubuntu-1
spec:
  running: true
  template:
    metadata:
      labels:
        special: vmi-macvtap
    spec:
      domain:
        devices:
          disks:
            - name: containerdisk
              disk:
                bus: virtio
            - name: cloudinitdisk
              disk:
                bus: virtio
          interfaces:
          - name: default
            masquerade: {}
          - name: l2-network
            macvtap: {}
        machine:
          type: ""
        resources:
          requests:
            memory: 1024M
      networks:
      - name: default
        pod: {}
      - name: l2-network
        multus: # Secondary multus network
          networkName: net1
          
      terminationGracePeriodSeconds: 0
      volumes:
        - name: containerdisk
          containerDisk:
            image: quay.io/containerdisks/ubuntu:22.04
        - name: cloudinitdisk
          cloudInitNoCloud:
            networkData: |
              version: 2
              ethernets:
                enp1s0:
                  dhcp4: true
                enp2s0:
                  addresses:
                    - 10.0.1.2/24
            userData: |-
              #cloud-config
              password: ubuntu
              chpasswd: { expire: False }
              ssh_authorized_keys:
                - ssh-rsa <your key here>
              packages: 
                - qemu-guest-agent
                - lldpd
                - nmap
              runcmd:
                - [ systemctl, start, qemu-guest-agent]
---
apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachine
metadata:
  name: vm-ubuntu-2
spec:
  running: true
  template:
    metadata:
      labels:
        special: vmi-macvtap
    spec:
      domain:
        devices:
          disks:
            - name: containerdisk
              disk:
                bus: virtio
            - name: cloudinitdisk
              disk:
                bus: virtio
          interfaces:
          - name: default
            masquerade: {}
          - name: l2-network
            macvtap: {}
        machine:
          type: ""
        resources:
          requests:
            memory: 1024M
      networks:
      - name: default
        pod: {}
      - name: l2-network
        multus: # Secondary multus network
          networkName: net1

      terminationGracePeriodSeconds: 0
      volumes:
        - name: containerdisk
          containerDisk:
            image: quay.io/containerdisks/ubuntu:22.04
        - name: cloudinitdisk
          cloudInitNoCloud:
            networkData: |
              version: 2
              ethernets:
                enp1s0:
                  dhcp4: true
                enp2s0:
                  addresses:
                    - 10.0.1.3/24
            userData: |-
              #cloud-config
              password: ubuntu
              chpasswd: { expire: False }
              ssh_authorized_keys:
                - ssh-rsa <your key here>
              packages: 
                - qemu-guest-agent
                - lldpd
                - nmap
              runcmd:
                - [ systemctl, start, qemu-guest-agent]


```
</details>

**Comment by fizzers123** [comment]
I did some additional testing and found something interesting. 
TL;DR: LLDP (L2 connectivity) works when starting TCPdump in virt-launcher

When the VMs are successfully started they can ping each other no problem. Ping is possible on the same node and if the VMs are on different nodes. At the beginning the VMs don't see anyone with LLDP. The underlying hypervisor or networkswitch sees both VMs. The screenshot below is from the Proxmox (sentinel) host that hots the Kubernetes nodes. It can see both the VM (vm-ubuntu-1) and the Kubernetes node (node1). 

Proxmox (or core switch)
![macvtap-proxmox-lldpd](https://user-images.githubusercontent.com/15678530/227994402-6de541be-eb61-4f79-866e-3f7164f82a94.png)

Now the interessting part comes. To debug this behavior a tcpdump was starte in the virtlaunchers net1 interface. This tcpdump is running using the network namespace of that container. As soon as this tcpdump was running the VMs saw the Proxmox via LLDP and each other, as long as they are running on the same node.  
For both VMs to discover each other two tcpdumps will need to run. One for each VMs Net1 interface. S

![macvtap-node1-container-interfaces](https://user-images.githubusercontent.com/15678530/227994489-73b9122e-044a-4a81-8997-fe3e809be879.png)

An Easy way to find the correct network namespace is to list all interfaces using the following command:
``` 
ip link show; ip -all netns exec ip link show
```
![macvtap-node1-tcpdump](https://user-images.githubusercontent.com/15678530/227994515-dda3dc63-433b-4980-97a4-b45592e79860.png)

VM with tcpdump running on the virtlauncher net1 interface:


![macvtap-vm-ubuntu-1-lldpd](https://user-images.githubusercontent.com/15678530/227994563-0b20b857-3c6c-416c-8b3d-34b9b2157cd0.png)
As soon as the  tcpdumps are stopped the VMs stop receiving the LLDP packages from the network. 

Setup:
- OS: Ubuntu 22.04.2 LTS
- Kubernetes: Kubespray + Kubevirt 
- CNI: OVN + Multus + MacVtap
